tests/components/datadog/test_config_flow.py

```diff
@@ -0,0 +1,144 @@
+"""Tests for the Datadog config flow."""
+
+from unittest import mock
+from unittest.mock import patch
+
+from homeassistant.components import datadog
+from homeassistant.const import STATE_UNKNOWN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+from homeassistant.setup import async_setup_component
+
+from .common import MOCK_CONFIG
+
+from tests.common import EVENT_STATE_CHANGED, MockConfigEntry
+
+
+async def test_user_flow_success(hass: HomeAssistant) -> None:
+    """Test user-initiated config flow."""
+    with patch(
+        "homeassistant.components.datadog.config_flow.validate_datadog_connection",
+        return_value=True,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            datadog.DOMAIN, context={"source": "user"}
+        )
+        assert result["type"] == FlowResultType.FORM
+
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=MOCK_CONFIG
+        )
+        assert result2["type"] == FlowResultType.CREATE_ENTRY
+        assert result2["data"] == MOCK_CONFIG
+
+
+async def test_user_flow_cannot_connect(hass: HomeAssistant) -> None:
+    """Test connection failure."""
+    with patch(
+        "homeassistant.components.datadog.config_flow.validate_datadog_connection",
+        return_value=False,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            datadog.DOMAIN, context={"source": "user"}
+        )
+
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=MOCK_CONFIG
+        )
+        assert result2["type"] == FlowResultType.FORM
```

Comment:
Tests should end with `CREATE_ENTRY` or `ABORT`.
Please continue this test with a successful connection so it ends with `CREATE_ENTRY`